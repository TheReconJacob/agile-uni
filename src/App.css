.searchContainer {
  display: flex;
  flex: 10;
  flex-direction: row;
}

.search {
  flex: 5;
}
