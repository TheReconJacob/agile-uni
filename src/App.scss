@import "sky-toolkit-core/all";
@import "sky-toolkit-ui/all";
@import "~@sky-uk/molecules/lib/SimpleMasthead";
@import "~sky-toolkit-ui/components/links";
@import "~sky-toolkit-ui/components/forms";
@import "~sky-toolkit-ui/components/buttons";
@import "~sky-toolkit-ui/components/typography"; // if using the `message` prop
@import "@sky-uk/toolkit-react/lib/ControlMessage"; // if using the `message` prop
@import "~sky-toolkit-ui/components/dropdown";
@import '~sky-toolkit-ui/components/hero';
@import '~sky-toolkit-ui/components/shine';
@import '~sky-toolkit-ui/components/typography';
@import '~sky-toolkit-ui/components/accordion';
@import '@sky-uk/toolkit-react/lib/Accordion';

.wrapper{
    min-height: calc(93vh - 72px);
    // min-height:100vh;
}