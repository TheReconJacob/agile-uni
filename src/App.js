import React from "react";
import "./App.scss";
import { AzureAD } from "react-aad-msal";
import { authProvider } from "./authProvider";
import Courses from "./pages/Courses";
import Home from "./pages/Home";
import Navbar from "./components/Navbar.js"
import Footer from "./components/Footer.js"
import { Route, BrowserRouter as Router } from "react-router-dom";
import Admin from "./pages/Admin";

const jwt = require('jsonwebtoken');

class App extends React.Component {
  constructor(props) {
    super(props);
    // We can add more state
    this.state = {
      admin: false,
    };

    this.getRoles().then(roles => {
      if (roles != null && roles.includes('admin')){
        this.setState({
          admin: true
        });
      }
    });
  }

  async getRoles(){
    const token = await authProvider.getIdToken();
    const idToken = token.idToken.rawIdToken;
    const decodedToken = jwt.decode(idToken);

    if (decodedToken.roles !== undefined) {
      console.log('Returning roles');
      return decodedToken.roles;
    } else {
      console.log('No specific role - general user')
      return null
    }
  }

  render() {
    let adminComponent;
    let adminAddCourse;

    if (this.state.admin) {
      adminComponent = <h1>HELLO ADMIN</h1>;
      adminAddCourse = <Route path="/admin" component={Admin} />;
    }
    return (
      <>
        <AzureAD provider={authProvider} forceLogin={true}>
          <Router>
            <Navbar />
            <div>
              <Route exact path="/" component={Home} />
              <Route path="/courses" render={() => <Courses adminStatus={this.state.admin} />} />
              {adminAddCourse}
            </div>
            {adminComponent}
            <Footer />
          </Router>
        </AzureAD>
      </>

    );
  }
}

export default App;
