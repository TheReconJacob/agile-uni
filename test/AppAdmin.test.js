import React from 'react';
import App from '../src/App.js';
import localStorageMock from '../src/localStorageMock'
import renderer from 'react-test-renderer';

beforeEach(() => {
    localStorageMock.setItem('msal.idtoken', 'eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsImtpZCI6IkJCOENlRlZxeWFHckdOdWVoSklpTDRkZmp6dyJ9.eyJhdWQiOiJjMGZiNzliYS1iNzJjLTQ3YzEtOTEyYy00OGVlNmNiYWM5NzIiLCJpc3MiOiJodHRwczovL2xvZ2luLm1pY3Jvc29mdG9ubGluZS5jb20vNjhiODY1ZDUtY2YxOC00YjJiLTgyYTQtYTRlZGRiOWM1MjM3L3YyLjAiLCJpYXQiOjE1NzMyMTg5NjUsIm5iZiI6MTU3MzIxODk2NSwiZXhwIjoxNTczMjIyODY1LCJhaW8iOiJBV1FBbS84TkFBQUFhZjNGZE13VksvOUVKaGdyYkc0M1J3YVhYMmF6cDB0dVU2V1k3SVdERUlHTmRkR0tCNlRlVWd4UUhabUZtN2dRNkF5N1VMNnhzSStFTHlqSnVPZmJsS0h4dktQdEhod1F6SFQvUzNQVWU4WnBuUG81Y2YxY0lwcUM2RGVERW1SOCIsIm5hbWUiOiJLZXJyeSwgU2hlaWxhIChBc3NvY2lhdGUgU29mdHdhcmUgRGV2ZWxvcGVyKSIsIm5vbmNlIjoiMTMyNDZkMTgtMGNlNy00MTZmLTllMzAtNGFkZGIyNGY0MTRhIiwib2lkIjoiYTEwMzQxYmUtMjRjYS00NWY5LWEyYWEtZjEyNzRjZjI2ZjhiIiwicHJlZmVycmVkX3VzZXJuYW1lIjoic2hlaWxhLmtlcnJ5QHNreS51ayIsInJvbGVzIjpbImFkbWluIl0sInN1YiI6IkJWeV9oN0wwZnBGamtXZ3dJaTVmd2NhYzZkYVM2QmZvNlJOZGF5a3hNYjAiLCJ0aWQiOiI2OGI4NjVkNS1jZjE4LTRiMmItODJhNC1hNGVkZGI5YzUyMzciLCJ1dGkiOiJ6aDdIcTRUNldraU85WWFqemI0LUFBIiwidmVyIjoiMi4wIn0.JWVqGskgxyBnn5NcGgNnPanDmpvd19LBXmz457gJvTnpsOgX-uXMrYPaGjMM6rKfQkAcgFkx6TmN26jnr3O3sG7KTgPs_4G6wXh3ZZf4px1LCwsBnKlHoznoIS26M88uZPGBZfHq7sFGH_80kCtqlwom7_0xsH0gpHi0TvRYLrilTyjVBQPX6ONg8U2art1d1Bu_DOt33I6d74hpvFpyG0Mj8Uw9HCzo0TT_OurShhZPU3QgDMDoBIUGYHsSOjkX7hk_jQKoOQbif2G3dOPOd02RiMtrSmYmoyJAVsK-jINF3kySNeYSvPlG3S-1nSAGrLAmk_MKHRNBsfxV5PIr6A')

});

test('Admin component only visible to admin', () => {
    const component = renderer.create(
        <App />
    )
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});