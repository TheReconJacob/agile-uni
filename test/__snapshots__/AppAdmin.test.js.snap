// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin component only visible to admin 1`] = `null`;
